client/src/components/Button/Button.js

```diff
@@ -1,30 +1,34 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import LoadingSpinner from 'components/LoadingSpinner/LoadingSpinner';
 import styles from 'components/Button/Button.module.scss';
 
-const Button = ({ text, disabled, btnRole = '' }) => {
+const Button = ({ text, isLoading = false, btnRole = '', ...otherProps }) => {
   const classes = btnRole
     .split(' ')
     .map((btnClass) => styles[btnClass])
     .join(' ');
 
   return (
     <div className={`${styles.btn} ${classes}`}>
-      <button type="submit" disabled={disabled}>
-        {text}
+      <button type="submit" {...otherProps}>
+        {isLoading ? <LoadingSpinner /> : text}
       </button>
     </div>
   );
 };
 
 Button.defaultProps = {
-  btnRole: ''
+  btnRole: '',
+  isLoading: false,
+  disabled: false
 };
 
 Button.propTypes = {
   text: PropTypes.string.isRequired,
-  disabled: PropTypes.func.isRequired,
-  btnRole: PropTypes.string
+  disabled: PropTypes.bool,
+  btnRole: PropTypes.string,
+  isLoading: PropTypes.bool
 };
```

Comment:
use props.children instead of passing the text as a prop

```
const Button = ({ children , isLoading = false, btnRole = '', ...otherProps }) => {
  const classes = btnRole
    .split(' ')
    .map((btnClass) => styles[btnClass])
    .join(' ');

  return (
    <div className={`${styles.btn} ${classes}`}>
      <button type="submit" {...otherProps}>
        {isLoading ? <LoadingSpinner /> : children}
      </button>
    </div>
  );
};


<Button>I am the button's child!</Button>

```

the propTypes for props.children is: 

```
children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node
    ]).isRequired


```